lms/djangoapps/course_api/blocks/serializers.py

```diff
@@ -67,6 +72,23 @@ def to_representation(self, block_key):
                     supported_field.default_value,
                 )
                 if field_value is not None:
+
+                    # override due date with student extension in unit added by instructor.
+                    if supported_field.block_field_name is 'due':
+                        try:
+                            override = StudentFieldOverride.objects.get(
+                                course_id=block_key.course_key,
+                                student=self.context['request'].user,
+                                location=block_key,
+                                field='due'
+                            )
+                        except StudentFieldOverride.DoesNotExist:
```

Comment:
instead of `get`, can we use `filter().fisrt()` in that case we won't need to except this exception.